framework/framework.go

```diff
@@ -371,10 +371,6 @@ func ProcessDelete(ctx context.Context, obj interface{}, resources []Resource) e
 			if err != nil {
 				return microerror.Mask(err)
 			}
-
-			if patch == nil {
-				return microerror.Maskf(executionFailedError, "patch must not be nil")
-			}
```

Comment:
I changed this in `ProcessDelete` and `ProcessUpdate`. `ProcessCreate` is going to be replaced in another PR.